packages/plugin-commands-env/src/resolveNodeVersion.ts

```diff
@@ -1,15 +1,16 @@
 import { FetchFromRegistry } from '@pnpm/fetch'
 import semver from 'semver'
 import versionSelectorType from 'version-selector-type'
+import { DEFAULT_NODE_MIRROR } from './constants'
 
 interface NodeVersion {
   version: string
   lts: false | string
 }
 
-export default async function resolveNodeVersion (fetch: FetchFromRegistry, rawVersionSelector: string) {
+export default async function resolveNodeVersion (fetch: FetchFromRegistry, rawVersionSelector: string, nodeMirror: string = DEFAULT_NODE_MIRROR) {
   const { releaseDir, version } = parseNodeVersionSelector(rawVersionSelector)
-  const response = await fetch(`https://nodejs.org/download/${releaseDir}/index.json`)
+  const response = await fetch(`${nodeMirror}/${releaseDir}/index.json`)
```

Comment:
This PR doesn't currently test that the `resolveNodeVersion` function actually uses the new `node-mirror` option. This was a bit difficult to intercept without setting up Jest mocks for `@pnpm/fetch` or larger refactors.

Happy to do a `jest.mock("pnpm/fetch")` test to spy this line if that's desirable. For now I thought that was more complicated than it was worth.

Reply:
Yes, I think it should be tested. You may use a mock or nock. You can see how nock is used here: https://github.com/pnpm/pnpm/blob/main/packages/npm-resolver/test/index.ts

Reply:
Woah, I didn't know about `nock`. That makes this so easy. Thank you for telling me about it and linking to an example! 🙂